src/Analyzer.TemplateProcessor/ArmTemplateProcessor.cs

```diff
@@ -378,6 +426,12 @@ private TemplateResource[] ReorderResourceCopies(Template template, Dictionary<s
                         else if (!dropResourceCopies)
                         {
                             updatedOrder.Add(resource);
+                            resource.Path = $"resources[{updatedOrder.Count - 1}]";
```

Comment:
I'm realizing now this will break if there are 2 or more resources with their own copy loops.  Both will be added to the end by the parsing and expansion, then reordering will add mappings for the copies of the first resource (below), then the second resource with a copy loop will be inserted into the array above (with `CopyIndex == 0`), pushing out the previous copies an extra index, making their previous mappings invalid.